src/Queen.cpp

```diff
@@ -0,0 +1,17 @@
+#include "Queen.h"
+
+Queen::Queen(bool isWhite)
+	:Piece(isWhite)
+{
+}
+
+bool Queen::isCorrectMove(const Location& source, const Location& destination) const
+{
+	return (source.x == destination.x || source.y == destination.y) ||
+			(abs(destination.x - source.x) == abs(destination.y - source.y));

```

Comment:
you used abs to calculate the absolute difference between coordinates. Make sure to consistently use std::abs throughout the code to maintain consistency.